pegasus_wrapper/workflow.py

```diff
@@ -407,7 +787,166 @@ def run_python_on_args(
+            "bash",
+            str(path_to_bash.absolute()),
+            site=self._default_site,
+            container=container,
+            is_stageable=job_is_stageable,
+            bypass_staging=job_bypass_staging,
+        ).transformation
+
+        job_dir = self.directory_for(job_name)
+        ckpt_name = job_name / "___ckpt"
+        ckpt_path = job_dir / "___ckpt"
+        job_script = job_dir / "script.sh"
+
+        commands_with_ckpt = list(command)
+        commands_with_ckpt.append(f"touch {ckpt_path.absolute()}")
+
+        job_script.write_text("\n".join(commands_with_ckpt))
+
+        bash_job = Job(bash_transform)
+        bash_job.add_args(str(job_script.absolute()))
+        bash_job.add_dagman_profile(category=category, retry=str(times_to_retry_job))
+
+        for profile in job_profiles:
+            bash_job.add_profiles(profile.namespace, key=profile.key, value=profile.value)
+
+        resource_request = self.set_resource_request(resource_request)
+        resource_request.apply_to_job(bash_job, job_name=self._job_name_for(job_name))
+
+        self._job_graph.add_jobs(bash_job)
+        for parent_dependency in depends_on:
+            if parent_dependency.job:
+                self._job_graph.add_dependency(bash_job, parents=[parent_dependency.job])
+            for out_file in parent_dependency.output_files:
+                bash_job.add_inputs(out_file)
+
+        # Handle Output Files
+        # This is currently only handled as the checkpoint file
+        # See: https://github.com/isi-vista/vista-pegasus-wrapper/issues/25
+        # If the checkpoint file already exists, we want to add it to the replica catalog
+        # so that we don't run the job corresponding to the checkpoint file again
+        checkpoint_pegasus_file = self.create_file(
+            self._job_name_for(ckpt_name), ckpt_path, add_to_catalog=ckpt_path.exists()
+        )
+
+        bash_job.add_outputs(checkpoint_pegasus_file, stage_out=False)
+
+        dependency_node = DependencyNode.from_job(bash_job, output_files=None)
+
+        self._signature_to_job[signature] = dependency_node
+        logging.info("Scheduled bash job %s", job_name)
+
+        return dependency_node
+
+    def start_docker_as_service(
+        self,
+        container: Container,
+        *,
+        depends_on,
+        mounts: Union[Iterable[str], str] = immutableset(),
+        docker_args: str = "",
+        resource_request: Optional[ResourceRequest] = None,
+    ) -> DependencyNode:
+        """
+        Start a docker image as a service
+        """
+        if isinstance(mounts, str):
+            mounts = immutableset(mounts)
+
+        container_loc = Locator(("containers", container.name))
+        container_dir = self.directory_for(container_loc)
+        container_start_path = container_dir / "start.sh"
+        container_stop_path = container_dir / "stop.sh"
+
+        docker_args += " -v ".join(mounts)
```

Comment:
Has this been tested with multiple mounts?

Reply:
Yes, verified to work with multiple mounts.